### PR TITLE
raftstore-v2: support unsafe recovery create and destroy peer

### PR DESCRIPTION
**Pull request by overvenus**
### What is changed and how it works?

Issue Number: ref https://github.com/tikv/tikv/issues/15108

What's Changed:

```commit-message
* raftstore-v2: support unsafe recovery create peer

  Reuse split mechanism to create peer. Because in v2 the only way to
  create and initialize to peer is via a snapshot message.

* raftstore-v2: support unsafe recovery destroy peer 
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/raftstore-v2/src/operation/query/mod.rs [hunk 1]
```diff
@@ -141,13 +141,9 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
             ));
         }
 
-        // TODO: add flashback_state check
```
Comment (Connor1996) [inline components/raftstore-v2/src/operation/query/mod.rs:144]: not added

Comment (overvenus) [reply components/raftstore-v2/src/operation/query/mod.rs]: It's a clean up, flashback is supported in v2.

File: components/raftstore-v2/src/operation/life.rs [hunk 2]
```diff
@@ -340,7 +345,8 @@ impl Store {
         &mut self,
         ctx: &mut StoreContext<EK, ER, T>,
         msg: Box<RaftMessage>,
-    ) where
+    ) -> bool
```
Comment (tonyxuqqi) [inline components/raftstore-v2/src/operation/life.rs:348]: please add the comment on the return value

**Review by tonyxuqqi — APPROVED** [review]

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**